InkyPhat/app/src/main/java/com/novoda/inkyphat/ImageConverter.java

```diff
@@ -0,0 +1,94 @@
+package com.novoda.inkyphat;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Matrix;
+import android.graphics.Paint;
+import android.graphics.Typeface;
+
+import static android.graphics.Paint.ANTI_ALIAS_FLAG;
+import static com.novoda.inkyphat.InkyPhat.Orientation.PORTRAIT;
+
+class ImageConverter {
+
+    private final InkyPhat.Orientation orientation;
+    private final ImageScaler imageScaler;
+    private final ColorConverter colorConverter;
+
+    ImageConverter(InkyPhat.Orientation orientation) {
+        this.orientation = orientation;
+        this.imageScaler = new ImageScaler();
+        this.colorConverter = new ColorConverter();
+    }
+
+    InkyPhat.PaletteImage convertImage(Bitmap input, Matrix.ScaleToFit scaleToFit) {
+        return translateImage(filterImage(input, scaleToFit));
+    }
+
+    private InkyPhat.PaletteImage translateImage(Bitmap input) {
+        int width = input.getWidth();
+        int height = input.getHeight();
+        int[] pixels = new int[width * height];
+        input.getPixels(pixels, 0, width, 0, 0, width, height);
+        InkyPhat.Palette[] colors = new InkyPhat.Palette[width * height];
+        for (int i = 0, pixelsLength = pixels.length; i < pixelsLength; i++) {
+            colors[i] = colorConverter.convertARBG888Color(pixels[i]);
+        }
+        return new InkyPhat.PaletteImage(colors, width);
+    }
+
+    Bitmap filterImage(Bitmap sourceBitmap, Matrix.ScaleToFit scaleToFit) {
+        return scaleToInkyPhatBounds(sourceBitmap, scaleToFit);
+    }
+
+    private Bitmap scaleToInkyPhatBounds(Bitmap sourceBitmap, Matrix.ScaleToFit scaleType) {
+        int bitmapWidth = sourceBitmap.getWidth();
+        int bitmapHeight = sourceBitmap.getHeight();
+        if (bitmapWidth < getOrientatedWidth() && bitmapHeight < getOrientatedHeight()) {
+            return sourceBitmap;
+        }
+
+        switch (scaleType) {
+            case FILL:
+                return imageScaler.fitXY(sourceBitmap, getOrientatedWidth(), getOrientatedHeight());
+            case START:
+            case CENTER:
+            case END:
```

Comment:
perhaps this should be 2 different things:

 1. `ScaleType` (fitXY, crop, etc.)
 2. `Alignment` (start, center, end)